tempodb/search/backend_search_block.go

```diff
@@ -23,6 +26,101 @@ type BackendSearchBlock struct {
 	l        *local.Backend
 }
 
+//nolint:golint
+type SearchDataCombiner struct{}
+
+func (*SearchDataCombiner) Combine(_ string, searchData ...[]byte) ([]byte, bool) {
```

Comment:
it feels weird to mimic the Combiner interface and ignore the format string. Should the format string indicate that the thing being combined is marshalled search data and not object data?

Does it make sense to mimic the interface at all? It's an "ObjectCombiner" and these aren't objects. Are we conflating two things that happen to match in the function signature but are semantically different?

Reply:
Reuse of DedupingIterator is the main thing.  If we want to split this from object combination, I would lean towards making DedupingIterator take a func callback, instead of an interface thing. That seems simpler.

Reply:
This makes sense. I'm fine with either path. I think we should take into consideration the broader goal is likely to deeply integrate these pieces with the existing backend block. Your all's call.